api/pkg/cli/installer/action.go

```diff
@@ -41,36 +42,89 @@ var (
 	ErrSameVersion              = errors.New("resource already exists with the requested verion")
 	ErrLowerVersion             = errors.New("cannot upgrade resource as requested version is lower than existing")
 	ErrHigherVersion            = errors.New("cannot downgrade resource as requested version is higher than existing")
+	ErrVersionIncompatible      = errors.New("requires compatible version")
+	ErrWarnVersionNotFound      = errors.New("pipeline version unknown")
 )
 
 type action string
 
 const (
-	update    action = "update"
-	upgrade   action = "upgrade"
-	downgrade action = " downgrade"
+	update             action = "update"
+	upgrade            action = "upgrade"
+	downgrade          action = " downgrade"
+	resourceMinVersion string = "tekton.dev/pipelines.minVersion"
 )
 
+func (i *Installer) TektonPipelinesVersion() {
+
+	var err error
+	i.PipelineVersion, err = tknVer.GetPipelineVersion(i.cs.Dynamic())
+	if err != nil {
+		i.PipelineVersion = ""
+	}
+}
+
+func (i *Installer) GetPipelineVersion() string {
```

Comment:
This function is not going anything like case handling, error handling etc. We can directly use the values, no need to write the function